web/packages/extension/src/common.ts

```diff
@@ -32,6 +32,35 @@ class CheckboxOption implements OptionElement<boolean> {
     }
 }
 
+class MaxExecutionDuration_NumberInputOption
+    implements OptionElement<Duration>
+{
+    constructor(
+        private readonly numberInput: HTMLInputElement,
+        readonly label: HTMLLabelElement
+    ) {}
+
+    get input() {
+        return this.numberInput;
+    }
+
+    get value() {
+        return Math.max(1, Math.round(parseInt(this.numberInput.value)));
+    }
+
+    set value(value: Duration) {
+        let newValue = 0;
+        if (typeof value === "number") {
+            newValue = value;
+        } else {
+            // Ignoring the 'nanos' part from the old format type ObsoleteDuration.
+            newValue = value.secs;
+        }
+
+        this.numberInput.value = "" + newValue;
```

Comment:
nit: Subject to personal preference, but in my opinion `newValue.toString()` or `String(newValue)` are more concise.